### PR TITLE
Fix for password recovery error messages

### LINKED ISSUE #11743: Wrong Recovery Password Error Messages
**The bug**
If the user enter a non valid password, the error messages displayed are not corrects

**To Reproduce**
1. Log out
2. Use 'Password Recovery' for your customer account
3. Go to password recovery link
4. Insert an INVALID password and an EQUAL confirmation (eg: pass: 1234 confirmation: 1234)
5. The error returned is : "The password and its confirmation do not match" instead of "The password is not in a valid format."

The error does not present itself in admin area.

**Additional information**
PrestaShop version: 1.7.4
PHP version: 7.2

### PR DESCRIPTION
**Pull request by web-fu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Correct error reporting in FO password recovery
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | No
| Deprecations? | No
| Fixed ticket? | Fixes #11743.
| How to test?  | .

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11745)

### CODE REVIEW

File: controllers/front/PasswordController.php [hunk 1]
```diff
@@ -121,16 +121,29 @@ protected function changePassword()
...
-                || !(Tools::isSubmit('confirmation'))
-                || ($passwd = Tools::getValue('passwd')) !== ($confirmation = Tools::getValue('confirmation'))
-                || !Validate::isPasswd($passwd) || !Validate::isPasswd($confirmation)) {
-                // Check if passwords are here anyway, BUT does not match the password validation format
-                if (Tools::isSubmit('passwd') || Tools::isSubmit('confirmation')) {
-                    $this->errors[] = $this->trans('The password and its confirmation do not match.', array(), 'Shop.Notifications.Error');
+            if ($isSubmit = Tools::isSubmit('passwd')) {
+                // If password is submitted validate pass and confirmation
+                if (!$passwd = Tools::getValue('passwd')) {
+                    $this->errors[] = $this->trans("The password is missing: please enter your new password.", array(), 'Shop.Notifications.Error');
```
Comment (LouiseBonnard) [inline controllers/front/PasswordController.php]: It is already localized in _Admin.Login.Notification_. Can a wording for the FO can be localized in the BO files? If yes, let it be attached to the _Admin.Login.Notification_ file. If no, I'm okay with this suggestion of localizing it in _Shop.Notifications.Error_. :-)

File: controllers/front/PasswordController.php [hunk 2]
```diff
@@ -121,16 +121,29 @@ protected function changePassword()
...
                 }
 
+                if (!$confirmation = Tools::getValue('confirmation')) {
+                    $this->errors[] = $this->trans('The confirmation is empty: please fill in the password confirmation as well', array(), 'Shop.Notifications.Error');
```
Comment (LouiseBonnard) [inline controllers/front/PasswordController.php]: Same here!

File: controllers/front/PasswordController.php [hunk 3]
```diff
@@ -121,16 +121,29 @@ protected function changePassword()
...
+                }
+
+                if ($passwd && $confirmation) {
+                    if ($passwd !== $confirmation) {
+                        $this->errors[] = $this->trans('The password and its confirmation do not match.', array(), 'Shop.Notifications.Error');
```
Comment (LouiseBonnard) [inline controllers/front/PasswordController.php]: I also have the following wording, _The password and its confirmation do not match. Please double check both passwords._, in _Admin.Login.Notification_.

File: controllers/front/PasswordController.php [hunk 4]
```diff
@@ -121,16 +121,29 @@ protected function changePassword()
...
+                    }
+
+                    if (!Validate::isPasswd($passwd)) {
+                        $this->errors[] = $this->trans('The password is not in a valid format.', array(), 'Shop.Notifications.Error');
```
Comment (LouiseBonnard) [inline controllers/front/PasswordController.php:138]: Same, only mentioned in _Admin.Login.Notification_.

File: controllers/front/PasswordController.php (hunk 1 above)
Comment (web-fu) [reply controllers/front/PasswordController.php]: The very best way should be abstract the password recover management at higher level: it's the same function in FO and in BO.
If you look at the chunk, it was written with some redundancy  3 years ago

> [*] Security : Add "change password" screen accessible from a "forgot password" query sent by mail. Part of #PSCSX-5770 for Admin only.  (cherry picked from commit 56133c5b34ab88e1e99c3a6fdce27b065be00d1e)

It's my first contribution on this project so I don't know what is the policy on small fix / feature redesign.

I think the unification of FO / BO password recover should be a feature for a next release

Comment (LouiseBonnard) [reply controllers/front/PasswordController.php]: @eternoendless, what's your opinion about it?

Comment (web-fu) [reply controllers/front/PasswordController.php]: What happens now?

Comment (eternoendless) [reply controllers/front/PasswordController.php]: > Can a wording for the FO can be localized in the BO files?

I don't think we should mix translations from FO and BO, that's part of the point of having translation domains, isn't it?

> I think the unification of FO / BO password recover should be a feature for a next release

I'm in favor of unifying some of the logic later on, but we should first list the functional differences (if any) between the two processes. Let's also keep in mind that wording domains should probably kept separate, as I said before.

**Review by eternoendless — DISMISSED** [review]
Looks good to me

File: controllers/front/PasswordController.php (hunk 1 above)
Comment (LouiseBonnard) [reply controllers/front/PasswordController.php]: Right, thanks @eternoendless. @web-fu, let's keep the _Shop.Notifications.Error_ domain then!

File: controllers/front/PasswordController.php (hunk 4 above)
Comment (LouiseBonnard) [reply controllers/front/PasswordController.php:138]: What about `Invalid password` in _Shop.Notifications.Error_?

**Review by eternoendless — APPROVED** [review]

File: controllers/front/PasswordController.php (hunk 4 above)
Comment (eternoendless) [reply controllers/front/PasswordController.php:138]: I suggest keeping the proposed wording:
- it's slightly clearer about the nature of the problem (even though it should be improved to explain what the expected requirements are)
- it's the same string that we use in the admin part

**Comment by Quetzacoalt91** [comment]
Thank you @web-fu

**Pull request merged**